src/lazy/expanded/macro_evaluator.rs

```diff
@@ -143,7 +143,7 @@ pub struct MacroEvaluator<
 > {
     // A stack of all of the macro invocations currently being evaluated.
     macro_stack: S,
-    spooky: PhantomData<(&'data D, &'data M)>,
+    spooky: PhantomData<(&'data D, M)>,
```

Comment:
I needed `'data` to appear somewhere in `MacroEvaluator` because it's being used in the type's trait bounds. I can leave it with `D` because `D` is guaranteed to outlive `'data`, but I couldn't have it with `M` because the types that represent a macro invocation include `'top` lifetime template values. (`&'top Element`)